react/features/conference/middleware.js

```diff
@@ -48,3 +52,33 @@ MiddlewareRegistry.register(store => next => action => {
 
     return result;
 });
+
+/**
+ * Set up state change listener to perform maintenance tasks when the conference
+ * is left or failed, close all dialogs and unpin any pinned participants.
+ */
+StateListenerRegistry.register(
+    state => getCurrentConference(state),
+    (conference, { dispatch, getState }, prevConference) => {
+        const { authRequired, passwordRequired }
+            = getState()['features/base/conference'];
+
+        if (conference !== prevConference) {
+            // Unpin participant, in order to avoid the local participant
+            // remaining pinned, since it's not destroyed across runs.
+            dispatch(pinParticipant(null));
+
+            // XXX I wonder if there is a better way to do this. At this stage
+            // we do know what dialogs we want to keep but the list of those
+            // we want to hide is a lot longer. Thus we take a bit of a shortcut
+            // and explicitly check.
+            if (typeof authRequired === 'undefined'
+                    && typeof passwordRequired === 'undefined'
+                    && !isDialogOpen(getState(), FeedbackDialog)) {
```

Comment:
This seems like a working solution, I'm not sure if it is the nicest indeed. But I can live with it for now.

@virtuacoplenny ?